### PR TITLE
Fix resample/interpolate for non-upsampling case

### LINKED ISSUE #1605: Resample interpolate failing on tutorial dataset
I'm getting some unexpected behavior/errors from the new resample/interpolate methods. 

@darothen - any idea what's going on here?

```Python-traceback
In [1]: import xarray as xr

In [2]: ds = xr.tutorial.load_dataset('air_temperature')

In [3]: ds.resample(time='15d').interpolate(kind='linear')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-ef931d7ebbda> in <module>()
----> 1 ds.resample(time='15d').interpolate(kind='linear')

/glade/p/work/jhamman/storylines/src/xarray/xarray/core/resample.py in interpolate(self, kind)
    110
    111         """
--> 112         return self._interpolate(kind=kind)
    113
    114     def _interpolate(self, kind='linear'):

/glade/p/work/jhamman/storylines/src/xarray/xarray/core/resample.py in _interpolate(self, kind)
    312
    313         old_times = self._obj[self._dim].astype(float)
--> 314         new_times = self._full_index.values.astype(float)
    315
    316         data_vars = OrderedDict()

AttributeError: 'NoneType' object has no attribute 'values'
```

### PR DESCRIPTION
**Pull request by jhamman**
- [x] Closes #1605
 - [x] Tests added / passed
 - [x] Passes ``git diff upstream/master | flake8 --diff``
 - ~[ ] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API~

cc @darothen

### CODE REVIEW

**Comment by darothen** [comment]
Great catch; do you need any input from me @jhamman ?

**Comment by jhamman** [comment]
@darothen - if this makes sense to you, we can let @shoyer give the final 👍 .

File: xarray/tests/test_dataarray.py [hunk 1]
```diff
@@ -2077,6 +2077,15 @@ def test_upsample_interpolate(self):
...
             self.assertDataArrayAllClose(expected, actual, rtol=1e-16)
 
+    @requires_scipy
+    def test_upsample_interpolate_regression_1605(self):
+        dates = pd.date_range('2016-01-01', '2016-03-31', freq='1D')
+        expected = xr.DataArray(np.random.random((len(dates), 2, 3)),
+                                dims=('time', 'x', 'y'),
+                                coords={'time': dates})
+        actual = expected.resample(time='1D').interpolate('linear')
+        self.assertDataArrayAllClose(actual, expected, rtol=1e-16)
```
Comment (MaximilianR) [inline xarray/tests/test_dataarray.py]: @jhamman reminder to use the assert functions rather than `self. ...` methods, for future PRs

**Comment by jhamman** [comment]
yes, this was a nice and easy fix. In the future, it may be nice to fast track no-op resample/interpolation since this does end up doing the same computation as it would if there were actual missing values but, this is a nice improvement for now.

Thanks again @darothen for the upsampling and resample refactor.

**Comment by darothen** [comment]
I'm a bit slow on the uptake here, but big 👍  from me. Thanks for catching this bug!

**Pull request merged**